views/windows/featureedit.html

```diff
@@ -16,6 +16,17 @@
           <simplemde text="ctrl.description" edit="true" is-required="false" maxlength="4000"></simplemde>
         </div>
       </div>
+
+      <div ng-if="ctrl.customFoldersEnabled" class="d-flex flex-row form-group">
+        <label class="col-2 col-form-label text-right">Folder</label>
+        <div class="col mt-2">
+          <select class="custom-select" ng-model="ctrl.folder"
+            ng-options="folder.label for folder in ctrl.folderOptions"
+            ng-change="ctrl.updateFolder()">
```

Comment:
This style of dropdown eliminates the tree structure of the folders and puts them in a flat list. Can we present the tree instead?

Reply:
In other places we also present a flattened tree for folder-only structures (such as managing favorites), so I would prefer to keep this consistent with those choices. If we decide to change how we present those options in the future we may want to change them all.

Reply:
I'll just go on record as stating that I disagree with that approach because two folders can have the same name but under different parent folders, and it would be difficult to then distinguish them from one another in a flat list.

Reply:
Completely agree with Will here. Saved Places is not a good candidate for displaying a flattened list, both because we often see folders with the same name and because the tree can get quite large and a list view would be easy to navigate/understand.

Reply:
The standard behavior for this kind of thing is most comparable to the Bookmark UI present in browsers. Chrome flattens the list and repeatedly shows identical folder names which, if confusing, is the user's fault or the code's fault. I know we just implemented something that will create duplicate folder names @schmidtk, but it would be better for it to try to number them.

Tree structures are not browsable, we don't have a tree search that maintains tree structure, and I think we should do everything we can to not show trees anywhere ever. The context of the tree being totally devoid of leaf nodes would also look very strange in my opinion. They suck if the tree becomes too deep because it's hard to even show all of the folder depth in the available horizontal space. We've been trying for years to get rid of the Add Data window for exactly these sorts of reasons.

I think a select2 is a better solution to the problem of a large tree structure and quickly finding the folder you want.

Reply:
Chrome presents a list of commonly used folders, along with a "Choose another folder" option that shows you the tree structure. This accommodates users with a lot of bookmarks, who may not find what they want in the limited initial list. Currently if you have a lot of places, you get something like this:
<img width="662" alt="Folders" src="https://user-images.githubusercontent.com/5685437/58647727-666b7f80-82c5-11e9-939c-1d5f932d7332.png">
I know we have folks with _way_ more places than that, making the list even more difficult to use.

A select2 would at least give them a way to search, but it would still present the duplicate name problem. That's not entirely in their control, because we allow saving an entire KML tree from an imported file.

Reply:
Another option might be adding spacers/tabs in to the front of the displayed text based on the depth of the folder. This could simulate the nesting structure without having any weirdness with the tree search or with collapsing/expanding nodes. It would also avoid having to build a whole custom UI (though would still have the problem with very deep nesting being terrible looking or unusable).